packages/freezed/lib/src/templates/concrete_template.dart

```diff
@@ -370,41 +367,123 @@ class Property {
       );
     }
 
+    final jsonKeyValues = element.jsonKeyValues;
+    if (defaultValue != null &&
+        jsonKeyValues != null &&
+        jsonKeyValues.containsKey('defaultValue')) {
+      throw InvalidGenerationSourceError(
+        '@Default cannot be used in conjunction with @JsonKey(defaultValue:)',
+        todo: 'Remove the defaultValue parameter from @JsonKey and only'
+            'use the @Default decorator instead',
+        element: element,
+      );
+    }
+
+    if (element.hasRequired &&
+        jsonKeyValues != null &&
+        jsonKeyValues.containsKey('required')) {
+      throw InvalidGenerationSourceError(
```

Comment:
Same here, why is this needed?

Reply:
This is because you could technically write something like this which would also be unclear in terms of what the outcome will be
```dart
factory User({@required @JsonKey(required: false) int age})
```